scipy/integrate/_ivp/ivp.py

```diff
@@ -492,6 +500,9 @@ def solve_ivp(fun, t_span, y0, method='RK45', t_eval=None, dense_output=False,
     events, is_terminal, event_dir = prepare_events(events)
 
     if events is not None:
+        if args is not None:
+            original_events = events
+            events = [lambda t, x: event(t, x, *args) for event in original_events]
```

Comment:
I think there is going to be an issue here. I think you'll run into this unintuitive thing when you define lambdas in a loop:
https://stackoverflow.com/questions/19837486/python-lambda-in-a-loop
The post contains the simple fix.

Reply:
Thanks. I thought about the late binding closures *gotcha* when I wrote that, but after I tried a variation of my demonstration script that had two event functions and it worked as expected, I put off thinking more about it.  I can create a problem with some simple examples in an ipython shell, so now I have to figure out why my demo worked (or why I *thought* it worked).

Reply:
I figured out why my example with two event functions appeared to work.  The functions returned the same value, but they had different directions.   Because the direction information has already been separated from the functions at this point in the code, my demo worked, even though both events were actually using the same function (the second one).

Reply:
A different example would be better for the tests : )

Reply:
Fixed.